cli/popper/utils.py

```diff
@@ -96,21 +96,44 @@ def is_popperized():
     return os.path.isfile(config_filename)
 
 
-def update_config(name, stages, envs, relative_path):
-    """Updates the configuration for a pipeline"""
-
-    if name == 'paper':
-        stages = 'build'
+def update_config(name, stages='', envs='', vars=[], relative_path=''):
+    """Updates the configuration for a pipeline."""
 
     config = read_config()
+    try:
+        pipeline_config = config['pipelines'][name]
+
+        if name == 'paper':
+            stages = 'build'
+        else:
+            if not stages:
+                stages = ','.join(pipeline_config['stages'])
+            if not envs:
+                envs = ','.join(pipeline_config['envs'])
+            if not relative_path:
+                relative_path = pipeline_config['path']
+    except KeyError:  # Pipeline does not exist yet
+        pass
+
     config['pipelines'][name] = {
         'stages': stages.split(','),
         'envs': envs.split(','),
+        'vars': vars,
         'path': relative_path
     }
     write_config(config)
 
 
+def read_pipeline_config(name):
+    """Returns the configuration for a pipeline."""
+
+    config = read_config()
+    try:
+        return config['pipelines'][name]
+    except KeyError:
+        fail("Pipeline {} does not exist.".format(name))
```

Comment:
This could return a 2-item tuple (`return config, pipeline_config`), so that we don't need to read the config separately.